tests/unit/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# @file      CMakeLists.txt
+# @brief     Unit Tests for Mender MCU platform independent code
+#
+# Copyright Northern.tech AS
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cmake_minimum_required(VERSION 3.16.3)
+project(mender-mcu-tests-unit)
+
+# TODO: move somewhere shared with smoke tests and remove the other fetch method
+include(FetchContent)
+FetchContent_Declare(
+  cjson
+  GIT_REPOSITORY https://github.com/DaveGamble/cJSON
+  GIT_TAG        v1.7.15
+  DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+)
+
+set(ENABLE_CJSON_TEST OFF CACHE INTERNAL "Turn off tests")
+FetchContent_MakeAvailable(cjson)
```

Comment:
I haven't decided yet, thanks for bringing it up.

We should either expect system provided libraries or fetch them, but not both. The old code already had the flaw that in the top `CMakeLists.txt` would use the system ones, while the `tests/CMakeLists.txt` one (now in `tests/smoke/...`) would fetch them from the "mocks" (I am talking plural because the same is true for MbedTLS). I think then that depending on how you run your CMake command you will get one or the other. Another related flaw was the (ab)use of `include` function on `CMakeLists.txt`.

These issues are still present for the smoke tests build.

I think we should flavor system ones in this repository, and unify the checks for it in the `target/posix/CMakeLists.txt`.

Reply:
I'll follow-up in the next PR